### PR TITLE
Cargo: Improve the OR-mapping of licenses

### PR DESCRIPTION
**Pull request by sschuberth**
Do not include the OR-mapping as part of the declared license strings, but
apply it when processing the declared licenses. This way licenses are
sorted alphabetically in the result, and it becomes clear from the
whether the original metadata used a "/" to separate licenses.

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

### CODE REVIEW

**Review by oheger-bosch — COMMENTED** [review]
The changes look good to me. I assume that the order in which licenses appear in an OR construct is irrelevant. There is, however, now a discrepancy in the license displayed by [crates.io](https://crates.io/crates/rand) for the _rand_ library (which is `MIT OR Apache-2.0`) and the one extracted by ORT (which is `Apache-2.0 OR MIT`).

File: analyzer/src/funTest/assets/projects/synthetic/cargo-expected-output.yml [hunk 1]
```diff
@@ -6,9 +6,10 @@ project:
   - "Another Author"
```
[the remark this answers is not in the document]
Comment (sschuberth) [reply analyzer/src/funTest/assets/projects/synthetic/cargo-expected-output.yml:6]: Thanks, I've fixed the sentence.

**Review by oheger-bosch — APPROVED** [review]

**Comment by sschuberth** [comment]
> I assume that the order in which licenses appear in an OR construct is irrelevant.

Correct.

> There is, however, now a discrepancy in the license displayed by [crates.io](https://crates.io/crates/rand) for the _rand_ library (which is `MIT OR Apache-2.0`) and the one extracted by ORT (which is `Apache-2.0 OR MIT`).

Crates.io displays the license for the latest version 0.8.3 by default. In that version, the [metadata contains an explicit OR](https://github.com/rust-random/rand/blob/6ecbe2626b2cc6110a25c97b1702b347574febc7/Cargo.toml#L5), whereas version 0.6.5 used in the test [still uses a slash](https://github.com/rust-random/rand/blob/0.6.5/Cargo.toml#L5). If ORT was coming across the version 0.8.3 instead, it should capture "MIT OR Apache-2.0" as the declared license and "Apache-2.0 OR MIT" as the (normalized) processed declared license.

**Comment by sschuberth** [comment]
> If ORT was coming across the version 0.8.3 instead, it should capture "MIT OR Apache-2.0" as the declared license and "Apache-2.0 OR MIT" as the (normalized) processed declared license.

It's almost like I said, looks like currently the order of licenses in an expression it not normalized:

```
- id: "Crate::rand:0.8.3"
  purl: "pkg:cargo/rand@0.8.3"
  authors:
  - "The Rand Project Developers"
  - "The Rust Project Developers"
  declared_licenses:
  - "MIT OR Apache-2.0"
  declared_licenses_processed:
    spdx_expression: "MIT OR Apache-2.0"
```

**Comment by sschuberth** [comment]
> the order of licenses in an expression it not normalized

And here's a proposal how to fix that: https://github.com/oss-review-toolkit/ort/pull/3703

**Pull request merged**